src/cli/mod.rs

```diff
@@ -0,0 +1,239 @@
+mod paths;
+mod prove_and_verify;
+mod repl;
+
+use std::fs;
+use std::path::PathBuf;
+
+use anyhow::{bail, Result};
+
+use lurk::eval::lang::Coproc;
+use lurk::field::{LanguageField, LurkField};
+use lurk::store::Store;
+use lurk::z_data::{from_z_data, ZData};
+use lurk::z_store::ZStore;
+use pasta_curves::{pallas, vesta};
+
+use clap::{Args, Parser, Subcommand};
+use tap::TapOptional;
+
+// use self::prove_and_verify::verify_proof;
+use self::repl::Repl;
+
+const DEFAULT_FIELD: LanguageField = LanguageField::Pallas;
+const DEFAULT_LIMIT: usize = 100_000_000;
+const DEFAULT_RC: usize = 10;
+
+#[derive(Parser, Debug)]
+#[clap(version)]
+struct Cli {
+    #[clap(subcommand)]
+    command: Command,
+}
+
+#[derive(Subcommand, Debug)]
+enum Command {
+    /// Loads a file, processing forms sequentially ("load" can be elided)
+    Load(LoadArgs),
+    /// Enters Lurk's REPL environment ("repl" can be elided)
+    Repl(ReplArgs),
+    // /// Verifies a Lurk proof
+    // Verify(VerifyArgs),
+}
+
+#[derive(Args, Debug)]
```

Comment:
Do we need the duplication between `LoadArgs` and `LoadCli`, or could we parse into a single `Args` struct?

Reply:
`Parse` and `Args` are incompatible deriving handlers. This is a workaround to support `lurk run file.lurk` and `lurk file.lurk` with the same parameters and behavior (analogously for `lurk repl` and `lurk`).